src/robotoff.js

```diff
@@ -126,5 +126,33 @@ export default {
     return axios.get(
       `${ROBOTOFF_API_URL}/images/logos?logo_ids=${logoIds.join(",")}`
     )
+  },
+
+  getNutritionValueFromImage(productCode, language, imageOcrUrl) {
+
+    var ocrUrlSubString = imageOcrUrl.split("/");
+
+    let keyName = `nutrition_${language}`;
+    
+    return axios.get(
+      `${OFF_API_URL}/product/${productCode}.json?fields=product_name,images`
+
+    ).then(result => {
+
+      for (var key in result.data.product.images) {
+        if (Object.prototype.hasOwnProperty.call(result.data.product.images, keyName)) {
+            if (/nutrition_[a-z][a-z]/.test(keyName)) {
```

Comment:
You defined `keyName` as
```js
let keyName = `nutrition_${language}`;
```

So I don't get the point of the reason of the second condition

Reply:
As you can see this part of the code is not doing what you were expecting. The `imgid` is not the correct one. 

In the video bellow, I replace the json of the ocr to see the picture, and it is the packaging not nutrition.

Please verify intermediate steps; For example you can add a `console.log(link_to_ocr_picture)` to see on which picture you are looking for information

![demoBugNutrition](https://user-images.githubusercontent.com/45398769/160301878-3ec827c1-753d-4068-b4d7-760938a9871a.gif)

Reply:
I finally understood why there is a bug. You loop over `key` but compute `if` according to the value of `keyName`